src/main/kotlin/org/jmailen/gradle/kotlinter/pluginapplier/AndroidSourceSetApplier.kt

```diff
@@ -14,18 +14,19 @@ import org.jmailen.gradle.kotlinter.id
 internal object AndroidSourceSetApplier : SourceSetApplier {
 
     override fun applyToAll(project: Project, action: SourceSetAction) {
-        val android = project.extensions.findByName("android")
-        (android as? BaseExtension)?.let {
-            it.sourceSets.all { sourceSet ->
-                val id = sourceSet.name.id
-                action(id, project.provider { getKotlinFiles(project, sourceSet) })
-            }
+        val android = project.extensions.findByName("android") as? BaseExtension ?: return
+        android.sourceSets.configureEach { sourceSet ->
+            val id = sourceSet.name.id
+            action(id, project.provider { getKotlinFiles(project, sourceSet) })
         }
     }
 
-    private fun getKotlinFiles(project: Project, sourceSet: AndroidSourceSet) = sourceSet.java.srcDirs.map { dir ->
-        project.fileTree(dir) { it.include("**/*.kt") }
-    }.reduce { merged: FileTree, tree: ConfigurableFileTree ->
-        merged + tree
+    private fun getKotlinFiles(project: Project, sourceSet: AndroidSourceSet): FileTree? {
+        val javaSources = sourceSet.java.srcDirs
+        val kotlinSources = (sourceSet.kotlin as? com.android.build.gradle.api.AndroidSourceDirectorySet)?.srcDirs.orEmpty()
```

Comment:
`sourceSet.kotlin` is accessible only on AGP 7.0 and up. AFAIU this wil fail if consumer's project uses lower version. This line is also a fix for the linked issue.

Reply:
Yes, I guess we'll need a compatibility boundary as per my earlier comment.